dotnet-engine/UnleashEngine/Types.cs

```diff
@@ -0,0 +1,40 @@
+using System.Text.Json.Serialization;
+
+namespace Unleash;
+
+public class Context
+{
+    public string? UserId { get; set; }
+    public string? SessionId { get; set; }
+    public string? RemoteAddress { get; set; }
+    public string? Environment { get; set; }
+    public string? AppName { get; set; }
+    public string? CurrentTime { get; set; }
+    public Dictionary<string, string>? Properties { get; set; }
+}
+
+public class EngineResponse {
+    [JsonPropertyName("error_message")]
+    public string? ErrorMessage { get; set; }
+
+    [JsonPropertyName("status_code")]
+    public string? StatusCode { get; set; }
+}
+
+public class EngineResponse<TResult> : EngineResponse {
```

Comment:
My .NET knowledge is failing me here, I don't actually know what this class declaration does O.O